CIP-XXX/CIP-XXX.md

```diff
@@ -27,10 +27,53 @@ As part of this, we also describe the bit ordering within a `BuiltinByteString`,
 and provide some laws these operations should obey.
 
 ## Motivation: why is this CIP necessary?
+Currently, Plutus Core has a `BuiltinByteString` type, which is a packed array of bytes, similar to Haskell's ByteString.
+We want to implement bit wise operations on `BuiltinByteString` type. In otherwords, we want to achieve bit level granularity on the operations on `BuiltinByteString`.
+Some of the operations that we are interested in are, bit level OR, AND, NOT, or XOR. There are many added advantage of having bit level granularity which we will discuss in more detail later. Unfortunately, all oprations available on the `BuiltinByteString` works on the byte level granularity. So we need this CIP to bridge the gap.
+To explain the usefulness of this granulaity to manipulate `BuiltinByteString`, we use the following example of integer sets.
+#### Integer sets
+Integer sets, also known as bitmap, bitset, or bitvector is used to represent a set of natural numbers. Integer sets can be represented as a `BuiltinByteString` of length $N$, where $N\in \mathbb{N}$ is the maximum possible number in that set. We can represent membership of an element $k \leq N$ by setting the $k^{th}$ bit is set in that `BuiltinByteString`.  For example, assume $ x = \{1,3,4,5\}$ be an integer set. We can represent this set using a byte string of size $8$ as $01011100$. Note that the $1^{st}$, $3^{rd}$, $4^{th}$ and $5^{th}$ bits are set and rest are zero. There are many advantages when we represet a set using Integer set. 
+1. Representing the empty set and the universe. We can use the `0x00`  or $00000000$ to represent an empty set, and `0xFF` or $11111111$ as the universe that can be represetend using $8$ bit byte string. 
```

Comment:
This isn't quite accurate: the empty set (and universe) depends on the maximum number that the integer set can represent.